noir-projects/aztec-nr/aztec/src/macros/functions/utils.nr

```diff
@@ -436,9 +508,7 @@ pub(crate) comptime fn create_authorize_once_check(
     let nonce_check_quote = f"{nonce_arg_name_quoted} == 0".quoted_contents();
 
     let fn_call = if is_private {
-        // At this point, the original args of the fn have already been altered by the macro
-        // to include PrivateContextInputs, so we need to adjust the args_len accordingly.
-        let args_len = f.parameters().len() - 1;
+        let args_len = f.parameters().len();
```

Comment:
This was very hard to fix as I only got "incorrect number of return values from a foreign call" error and to isolate it to a relevant oracle and to the num args being incorrectly computed here took me 2 hours. Might be a skill issue tho

Reply:
Nah this was definitely bad. At least there was a comment.